src/main/resources/public/ts/model/OrderClient.ts

```diff
@@ -186,8 +238,70 @@ export class OrderClient implements Order  {
         return Number(price.toFixed(2));
     }
 
-    calculatePriceTTC (selectedOptions: boolean):number{
-        return  Number((this.calculatePriceHT(selectedOptions) * (100 + this.tax_amount) / 100).toFixed(2));
+    calculatePriceTTC(selectedOptions: boolean): number {
+        return (this.price_proposal)
+            ? this.price_proposal
+            : Number((this.calculatePriceHT(selectedOptions) * (100 + this.tax_amount) / 100).toFixed(2));
+    }
+
+    build(orderClientResponse: IOrderClientResponse): OrderClient {
+        this.amount = orderClientResponse.amount;
+        this.comment = orderClientResponse.comment;
+        this.creation_date = new Date(orderClientResponse.creation_date);
+        if (orderClientResponse.done_date)
+            this.done_date = new Date(orderClientResponse.done_date);
+        if (orderClientResponse.files && orderClientResponse.files.length > 0 && orderClientResponse.files[0] !== null)
+            this.files = orderClientResponse.files;
+        this.id = orderClientResponse.id;
+        this.id_campaign = orderClientResponse.id_campaign;
+        this.image = orderClientResponse.image;
+        this.name = orderClientResponse.name;
+        this.preference = orderClientResponse.preference;
+        this.price = orderClientResponse.price;
+        this.price_proposal = orderClientResponse.price_proposal;
+        this.rank = orderClientResponse.rank;
+        this.status = orderClientResponse.status;
+        this.summary = orderClientResponse.summary;
+        this.tax_amount = orderClientResponse.tax_amount;
+        //supplier
+        this.supplier_name = orderClientResponse.name_supplier;
+        if (orderClientResponse.options && orderClientResponse.options.length > 0 && orderClientResponse.options[0] !== null)
+            this.options = orderClientResponse.options.map(options => {
```

Comment:
typage `options ` dans ta map

tu pourrais aussi le faire en une ligne : 

```suggestion
            this.options = orderClientResponse.options.map((options: IOrderClientOptionResponse) => new OrderOptionClient().build(options));
```